lib/core/jdl_application_configuration_factory.js

```diff
@@ -36,6 +42,10 @@ function createApplicationConfigurationFromObject(configurationObject = {}) {
 }
 
 function createJDLConfigurationOption(name, value) {
+  if (!doesOptionExist(name, value)) {
+    logger.debug(`Unrecognised application option name: ${name}`);
+    return null;
+  }
```

Comment:
The issue with that is that the JDLAppConfFactory will try to set a nil option.
If it's nil, you have this snippet
```javascript
if (!option) {
    throw new Error('An option has to be passed to set an option.');
}
```
that will throw an error, for the right reasons.
So I suggest rewriting the `createApplicationConfigurationFromObject` function like this:
```javascript
function createApplicationConfigurationFromObject(configurationObject = {}) {
  const configuration = new JDLApplicationConfiguration();
  Object.keys(configurationObject).forEach(optionName => {
    const optionValue = configurationObject[optionName];
    if (!doesOptionExist(optionName, optionValue)) {
      logger.debug(`Unrecognized application option name and value: ${optionName} and ${optionValue}`);
      return;
    }
    configuration.setOption(createJDLConfigurationOption(optionName, optionValue));
  });
  return configuration;
}
```

Reply:
Plus, there must be tests for that

Reply:
@MathieuAA : If I may ask, that first snippet; 

https://github.com/jhipster/jhipster-core/blob/ea058ca4e1c8eadb067571e5d98cb4927b1ce5d7/lib/core/jdl_application.js#L32-L34

When does it gets called? Because in my tests (I tested running export-jdl command); I didn't see that being called. :smile:

Reply:
Ah, I think I've mistakenly removed the test on `jdl_application_configuration.spec.js` which I've added back. I've made the changes as you've suggested. Let me know if you see any further issues. :smile:

Reply:
Ah of course, I got confused. If you remove the throw statement, then it's okay.
You understood ;)